src/main/scala/ing/wbaa/druid/client/DruidAdvancedHttpClient.scala

```diff
@@ -77,27 +78,29 @@ class DruidAdvancedHttpClient private (
 
     val request = HttpRequest(HttpMethods.GET, uri = druidConfig.healthEndpoint)
 
-    val checksF = Future.sequence {
+    val checksF: Future[Iterable[(QueryHost, Boolean)]] = Future.sequence {
       brokerConnections.map {
         case (queryHost, flow) =>
-          val responsePromise = Promise[HttpResponse]()
-
-          Source
-            .single(request -> responsePromise)
-            .via(flow)
-            .runWith(Sink.head)
-            .flatMap {
-              case (Success(response), _) =>
-                Future {
-                  response.discardEntityBytes()
-                  queryHost -> (response.status == StatusCodes.OK)
-                }
-              case (Failure(ex), _) => Future.failed(ex)
+          healthLogger.debug(
+            s"healthcheck of ${queryHost.host}:${queryHost.port} started: $request"
+          )
+          executeRequest(flow)(request)
+            .map { response =>
+              healthLogger.debug(
+                s"healthcheck of ${queryHost.host}:${queryHost.port} success: ${response.status}"
+              )
+              response.discardEntityBytes()
+              queryHost -> (response.status == StatusCodes.OK)
+            }
+            .recover {
+              case ex =>
+                healthLogger.info(
```

Comment:
I think it will be more appropriate to change the logging level to `error` (or `warning`), since the current host appears to be unhealthy.  Additionally, the throwable can be passed to a separate argument instead of interpolating to the message.

Reply:
I agree the `warn` level makes more sense (error seems a bit much, certainly if there's more than one broker) and I will change it accordingly.

However, logging the full stack trace here introduces a lot of noise. Even based on just the unit test output, I would prefer not to.